### PR TITLE
capture az error output and auto-fix the script

### PR DESCRIPTION
**Pull request by sulixu**
---
This PR enhance run script to capture error output, which we can feed to chatgpt to suggest corrections, etc

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az aks copilot

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aks-preview/azext_aks_preview/_openai_wrapper.py [hunk 1]
```diff
@@ -82,8 +86,70 @@ def run_system_script(script_content: str):
...
+
+
+def spawn_and_run_system_script(script_content: str):
+    """
+    Note: this doesn't work on Windows because there's no pty support
+    we use spawn to capture child output,
+    so user can still interact with yes/no answer from script
+    """
+    if IS_MS_WINDOWS:
+        cmd = ["powershell", "-Command", script_content]
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_openai_wrapper.py:133]: Seems this is never reached. Or according to description, should just return/raise exception?

Comment (sulixu) [reply src/aks-preview/azext_aks_preview/_openai_wrapper.py]: make sense, just add raise exception if this func is called from windows

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 2]
```diff
@@ -127,6 +128,25 @@ def assert_openai_interactive_shell_launched(self):
             if not pattern in output:
                 raise CliTestError(f"Output from aks copilot did not contain '{pattern}'. Output:\n{output}")
 
+    def test_detect_az_error(self):
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:131]: Could you put these test cases to some other file? The test cases in `test_aks_commands.py` are live test cases.

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**